### PR TITLE
CHK-1150-fix-decode-transactionId

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Handled http status 400 when decoding transactionId. 

#### Motivation and Context
Handled http status 400 when decoding transactionId. The service responded with 500

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/utils/UUIDUtils.java [hunk 1]
```diff
@@ -16,9 +20,14 @@ public String uuidToBase64(UUID uuid) {
...
-        ByteBuffer bb = ByteBuffer.wrap(bytes);
-        return new UUID(bb.getLong(), bb.getLong());
+    public Mono<UUID> uuidFromBase64(String str) {
+        try {
+            byte[] bytes = Base64.decodeBase64(str);
+            ByteBuffer bb = ByteBuffer.wrap(bytes);
+            return Mono.just(new UUID(bb.getLong(), bb.getLong()));
+        } catch (BufferUnderflowException e) {
+            return Mono.error(new InvalidRequestException("Error while decode transactionId"));
+        }
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/utils/UUIDUtils.java:30]: Given that it's not an asynchronous operation is it right to return `Mono`?

Comment (Fernando-Granato) [reply src/main/java/it/pagopa/transactions/utils/UUIDUtils.java]: You are right, I tried to solve it using an Either let me know what you think 😉

**Review by giovanniberti — COMMENTED** [review]
Add a test for an invalid byte array and it's on 🚀

File: src/test/java/it/pagopa/transactions/utils/UUIDUtilsTest.java [hunk 2]
```diff
@@ -27,4 +28,12 @@ void shouldEncodeAndDecodeBase64OfUUID() {
 
         assertEquals(uuid, uuidFromBase64.get());
     }
+
+    @Test
+    void shouldDecodeBase64OfUUIDError() {
+        String wrongUuid = "xxxx";
+        Either<InvalidRequestException, UUID> uuidFromBase64 = uuidUtils.uuidFromBase64(wrongUuid);
+        assertEquals(InvalidRequestException.class, uuidFromBase64.getLeft().getClass());
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/utils/UUIDUtilsTest.java:36]: This line is not actually needed as it is guaranteed from the type signature, a `assertTrue(uuidFromBase64.isLeft())` should suffice

Comment (Fernando-Granato) [reply src/test/java/it/pagopa/transactions/utils/UUIDUtilsTest.java]: you are right !!!! 👀

**Review by giovanniberti — APPROVED** [review]
🚀

**Pull request merged**